test/integration/query-interface/schemas.test.ts

```diff
@@ -36,4 +38,64 @@ describe('QueryInterface#createSchema', async () => {
 
     expect(queryGeneratorSpy.args[0]).to.include(options);
   });
+
+  it('drops a schema', async () => {
+    await queryInterface.createSchema(testSchema);
+    const preDeletionSchemas = await queryInterface.showAllSchemas();
+    expect(preDeletionSchemas).to.include(testSchema, 'createSchema did not create testSchema');
+
+    await queryInterface.dropSchema(testSchema);
+    const postDeletionSchemas = await queryInterface.showAllSchemas();
+    expect(postDeletionSchemas).to.not.include(testSchema, 'dropSchema did not drop testSchema');
+  });
+
+  it('shows all schemas', async () => {
+    await queryInterface.createSchema(testSchema);
+    const allSchemas = await queryInterface.showAllSchemas();
+
+    const expected = dialectsWithEqualDBsSchemas.includes(dialect.name)
+      ? [sequelize.config.database, testSchema]
+      : [testSchema];
+    expect(allSchemas.sort()).to.deep.eq(expected.sort());
+  });
+
+  describe('drops all schemas', () => {
+    it('drops all schemas except default schema', async () => {
+      await queryInterface.dropAllSchemas({
+        skip: [sequelize.config.database],
+      });
+      const postDeleteSchemas = await queryInterface.showAllSchemas();
+
+      const expected = dialectsWithEqualDBsSchemas.includes(dialect.name) ? [sequelize.config.database] : [];
+      expect(postDeleteSchemas).to.deep.eq(expected);
+    });
+
+    it('drops all schemas except test schema', async () => {
+      // required for testing this option with Postgres, MSSQL, DB2
+      if (!['postgres', 'mssql', 'db2'].includes(dialect.name)) {
+        return;
+      }
+
```

Comment:
```suggestion
```
In combination with suggestion below

Reply:
If that suggestion does not work you can also do this
```suggestion
      // 'skip' option is already tested above for these dialects
      if (dialectsWithEqualDBsSchemas.includes(dialect.name)) {
        return;
      }
```